### PR TITLE
Move menu icons up by 1px

### LINKED ISSUE #373: Menu icons should be moved up 1px
The + and - menu icons should be moved up 1px so that they line up better with the text: 

Before: 
Screen-Shot-2020-10-08-at-3 13 06-PM

After: 
Screen-Shot-2020-10-08-at-3 13 06-PM-new

### PR DESCRIPTION
**Pull request by aristath**
Fixes #373

## Summary
Changed the background-position of icons from `center center` to `calc(50% - 1px) center`.
Tested on Chrome & Firefox

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

**Review by GuidoOffermans — APPROVED** [review]
Quick and simple solution 👍

**Comment by aristath** [comment]
The previous solution had issues on some browsers (props @GuidoOffermans for testing it extremely thorougly)
left-to-right we've got firefox, safari, chrome:
Screenshot 2020-10-09 at 11 13 15

I had to refactor the way icons get added. The CSS is now a lot simpler. Icons get added inline as SVGs instead of backgrounds in buttons. It should be a lot more consistent and minimal.
If someone on OSX can test it that would be great :pray:

File: inc/menu-functions.php [hunk 1]
```diff
@@ -30,7 +30,11 @@ function twenty_twenty_one_add_sub_menu_toggle( $output, $item, $depth, $args )
 	if ( 0 === $depth && in_array( 'menu-item-has-children', $item->classes, true ) ) {
 
 		// Add toggle button.
-		$output .= '<button class="sub-menu-toggle" aria-expanded="false" onClick="twentytwentyoneExpandSubMenu(this)"><span class="screen-reader-text">' . esc_html__( 'Toggle child menu', 'twentytwentyone' ) . '</span></button>';
+		$output .= '<button class="sub-menu-toggle" aria-expanded="false" onClick="twentytwentyoneExpandSubMenu(this)">';
```
Comment (carolinan) [inline inc/menu-functions.php:33]: Should we make these icons available in the icon class again?
https://github.com/WordPress/twentytwentyone/blob/trunk/classes/class-twenty-twenty-one-svg-icons.php#L33

Comment (aristath) [reply inc/menu-functions.php:33]: I don't see why not... I'll push a commit with that in a minute

Comment (aristath) [reply inc/menu-functions.php:33]: Done

**Comment by kjellr** [comment]
> If someone on OSX can test it that would be great 🙏

This is unexpectedly complicated. 😄 

In Firefox, this looks close to correct. It's just slightly above the center: 

firefox

In Chrome + Safari though, it's appearing definitely too low still: 

safari

File: assets/sass/06-components/navigation.scss [hunk 2]
```diff
@@ -230,37 +230,43 @@
...
-			&[aria-expanded="false"] {
-				/* stylelint-disable */
-				background: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='18' height='18'><path d='M18 11.2h-5.2V6h-1.6v5.2H6v1.6h5.2V18h1.6v-5.2H18z' fill='%2328303d'/></svg>") no-repeat center center;
-				.is-background-dark & {
-					background: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='18' height='18'><path d='M18 11.2h-5.2V6h-1.6v5.2H6v1.6h5.2V18h1.6v-5.2H18z' fill='%23fff'/></svg>") no-repeat center center;
+			.icon-plus,
+			.icon-minus {
+				height: 100%;
+				display: flex;
+				align-items: center;
```
Comment (kjellr) [inline assets/sass/06-components/navigation.scss:236]: If I remove the `flexbox` + `height` rules, the appearance improves in Safari + Chrome: 

Before: 
safari

After: 
safari2

Unfortunately, this pushes it up too high in Firefox though: 

ff2

😩

**Comment by carolinan** [comment]
It doesn't have to be exact.... it is also an option to replace this icon...

**Comment by kjellr** [comment]
Yeah, @melchoyce would you give this PR a look and let us know what you think too? I _think_ it's better than [the version here](https://github.com/WordPress/twentytwentyone/pull/383#pullrequestreview-505909490), and we can always have another try at this during the betas if we decide to.

**Review by melchoyce — APPROVED** [review]
Yeah, I think this is a good enough solution. Thanks @aristath. We can merge once the conflicts are resolved.

**Pull request merged**